Task_2_3_2/src/main/java/r/masud/ccfit/nsu/ru/task_2_3_2/Main.java

```diff
@@ -0,0 +1,294 @@
+package r.masud.ccfit.nsu.ru.task_2_3_2;
+
+import javafx.animation.Animation;
+import javafx.animation.KeyFrame;
+import javafx.animation.Timeline;
+import javafx.application.Application;
+import javafx.event.EventHandler;
+import javafx.scene.Group;
+import javafx.scene.Scene;
+import javafx.scene.canvas.Canvas;
+import javafx.scene.canvas.GraphicsContext;
+import javafx.scene.image.Image;
+import javafx.scene.input.KeyCode;
+import javafx.scene.input.KeyEvent;
+import javafx.scene.paint.Color;
+import javafx.scene.text.Font;
+import javafx.stage.Stage;
+import javafx.util.Duration;
+
+
+
+import javax.print.attribute.standard.Media;
+import java.awt.Point;
+
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+public class Main extends Application {
+
+//    creating a frame
+    private static final int WIDTH = 600;
+    private static final int HEIGHT = WIDTH;
+    private static final int ROWS =30;
+    private static final int COLUMNS =ROWS;
+
+    private static final int SQUARE_SIZE = WIDTH/ROWS;
+    private static final String[] FOOD_ICONS = new String[]{
+            "C:\\Users\\Asus\\IdeaProjects\\oop\\Task_2_3_2\\src\\context\\carrot.png",
+            "C:\\Users\\Asus\\IdeaProjects\\oop\\Task_2_3_2\\src\\context\\mushroom.png",
+            "C:\\Users\\Asus\\IdeaProjects\\oop\\Task_2_3_2\\src\\context\\nut.png",
+            "C:\\Users\\Asus\\IdeaProjects\\oop\\Task_2_3_2\\src\\context\\sulphite.png"
+    };
+
+
+    private static final int RIGHT =0;
+    private static final int LEFT = 1;
+    private static final int UP = 2;
+    private static final int DOWN = 3;
+
+    private GraphicsContext gc;
+    private List<Point> snakeBody = new ArrayList();
+    private Point snakeHead;
+    private Image foodIcons;
+    private int foodX;
+    private int foodY;
+    private boolean gameOver;
+    private int currentDirection;
+    private int score = 0;
+
+
+
+
+
+
+
+    @Override
+    public void start(Stage primaryStage) throws IOException {
```

Comment:
You should implement MVP pattern, create separate View, Presenter and Model layers. Main should just start application.
And also there should be console version of view with appropriate presenter, along with JavaFX view